document_page_portal/static/src/js/document_page_portal_tour.js

```diff
@@ -0,0 +1,32 @@
+odoo.define('document_page_portal.tour', function (require) {
+    'use strict';
+
+    var tour = require("web_tour.tour");
+
+    tour.register('document_page_portal_tour', {
+        test: true,
+        url: '/my',
+    },
+    [
+        {
+            content: "Check document_page_portal is loaded",
+            trigger: 'a[href*="/my/knowledge/documents"]:contains("Knowledge Documents"):first',
+        },
+        {
+            content: "Check public document_page is loaded",
+            trigger: 'a[href*="/my/knowledge/documents/page"]:contains("Test Public Page 1"):first',
```

Comment:
I think the issue is here because. you're targeting the pager instead of the pages

https://github.com/OCA/knowledge/pull/239/files#diff-40f7e6ff05e04a9b73be7e0eae416c0e55a88c346c4c615d65a5807e72dd4f78R48

The qweb view is using this format: `/knowledge/document/#{document.id}`

So you should do something like this:

```suggestion
            trigger: 'a[href*="/knowledge/document/"]:contains("Test Public Page 1"):first',
```

Reply:
@ivantodorovich  Adjusted. It was this way before my update. However, I am having trouble reaching the code coverage goal

Reply:
By looking at the coverage report, it seems what's missing are testing search and date filters on the documents list.
But in my opinion this is ok to merge, you've done a good job ! 

If you still want to add the missing tests, you need to write the tours as if you were navigating the browser yourself. I think you can use this as an example: https://github.com/odoo/odoo/blob/338f8d843832c339c52c00c7f4c41b5ab2d696ed/addons/website_forum/static/tests/tours/website_forum_question.js#L29

This all means that, in your test, you have to mimic going to the documents list, typing some keywords in the searchbox, clicking on the search button, and check if the expected results are there. You can probably do all in the same tour, since you can move back and forth the different pages.

Reply:
@ivantodorovich Done 😄 .. thanks for your help. Ready to merge hehehe